envoy/network/connection.h

```diff
@@ -347,6 +347,13 @@ class Connection : public Event::DeferredDeletable,
    */
   virtual const ConnectionSocket::OptionsSharedPtr& socketOptions() const PURE;
 
+  /**
+   * Set a socket option on the underlying socket(s) of this connection.
+   * @param option The socket option to set.
+   * @return boolean telling if the socket option was set successfully.
+   */
+  virtual bool setSocketOption(const Network::Socket::OptionConstSharedPtr option) PURE;
```

Comment:
I think `Network::Socket::OptionConstSharedPtr` is the wrong type for this. The function is supposed to set the option now; this type is for storing an option and setting it later at the right time. I think something simpler, like `Option::Details` or similar, would be better here. Then we don't have to care about the state.

Reply:
I updated the interface to take an `Option::Details`.